pkg/yang/entry_test.go

```diff
@@ -1012,3 +1012,143 @@ func TestActionRPC(t *testing.T) {
 		}
 	}
 }
+
+// addTreeE takes an input Entry and appends it to a directory, keyed by path, to the Entry.
+// If the Entry has children, they are appended to the directory recursively. Used in test
+// cases where a path is to be referred to.
+func addTreeE(e *Entry, dir map[string]*Entry) {
+	for _, ch := range e.Dir {
+		dir[ch.Path()] = ch
+		if ch.Dir != nil {
+			addTreeE(ch, dir)
+		}
+	}
+}
+
+func TestEntryFind(t *testing.T) {
+	tests := []struct {
+		name            string
+		inModules       map[string]string
+		inBaseEntryPath string
+		wantEntryPath   map[string]string // keyed on path to find, with path expected as value.
+		wantError       string
+	}{{
+		name: "intra module find",
+		inModules: map[string]string{
+			"test.yang": `
+				module test {
+					prefix "t";
+					namespace "urn:t";
+
+					leaf a { type string; }
+					leaf b { type string; }
+
+					container c { leaf d { type string; } }
+				}
+			`,
+		},
+		inBaseEntryPath: "/test/a",
+		wantEntryPath: map[string]string{
+			// Absolute path with no prefixes.
+			"/b": "/test/b",
+			// Relative path with no prefixes.
+			"../b": "/test/b",
+			// Absolute path with prefixes.
+			"/t:b": "/test/b",
+			// Relative path with prefixes.
+			"../t:b": "/test/b",
+			// Find within a directory.
+			"/c/d": "/test/c/d",
+			// Find within a directory specified relatively.
+			"../c/d": "/test/c/d",
+			// Find within a relative directory with prefixes.
+			"../t:c/t:d": "/test/c/d",
+			"../t:c/d":   "/test/c/d",
+			"../c/t:d":   "/test/c/d",
+			// Find within an absolute directory with prefixes.
+			"/t:c/d": "/test/c/d",
+			"/c/t:d": "/test/c/d",
+		},
+	}, {
+		name: "inter-module find",
+		inModules: map[string]string{
+			"test.yang": `
+				module test {
+					prefix "t";
+					namespace "urn:t";
+
+					import foo { prefix foo; }
+					import bar { prefix baz; }
+
+					leaf ctx { type string; }
+					leaf other { type string; }
+				}`,
+			"foo.yang": `
+				module foo {
+					prefix "foo"; // matches the import above
+					namespace "urn:foo";
+
+					container bar {
+						leaf baz { type string; }
+					}
+				}`,
```

Comment:
What would happen if all modules contained a container or leaf of the same name.  For example, they all have "leaf conflict { type string }".  We should still be able to get to each of them by saying conflict, foo:confclit, bar:conflict, baz:conflict.